### ISSUE TITLE
Special characters in word list

### ISSUE THREAD

**Issue by richardfrost** [richardfrost/AdvancedProfanityFilter#64]
A user mentioned that they wanted to be able to remove a word that included an asterisk. I can think of two ways to fix this:

- Sanitize/escape input from add word (store it escaped)
- Sanitize when actually building the regex for the filter

**Comment by richardfrost** [comment]
Here is a way to simply escape input

```js
function escapeRegExp(str) {
  return str.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&");
}
```

**Comment by richardfrost** [comment]
I think if I escape special characters it would be good to also add an "expert" matching mode that doesn't do ANY processing and allows users to put a full regexp without any modification:

Considering how few people know regex (and even fewer who are good at it), I would suggest separating it into a separate "expert mode"  expanding section, with an explanation of the the basics. Even better would be a dropdown that lists what is trying to be found (for a few of the regex commands), and adds it for the user so they don't have to memorize the rules. You wouldn't get regex full power with a dropdown as with direct access, but it would be a compromise that I think many more people could handle.

**Comment by richardfrost** [comment]
This has been implemented in release `1.0.9`. By default all words/phrases containing a special character will be automatically escaped before generating the regular expression for the filter. The word/phrase is stored as a normal string exactly how you entered it.

There is also a new Regular Expression Match that can be applied on a per-word basis **([Advanced](https://github.com/richardfrost/AdvancedProfanityFilter/wiki#global-matching-method))**
